### PR TITLE
fix to default eval name, added log_params option for experiments

### PR DESCRIPTION
**Pull request by ericlee0803**
DEFAULT_EVALS_NAME right now is "Test Set", and no good because it has a space, which XGBoost isn't happy about (an error gets thrown). So I changed it to "TestSet". 

I also added an additional flag "log_params" to DEFAULT_RUN_OPTIONS, which turns off param logging. This is important for for experiments because we do not want to automatically log the params that are optimized (Sigopt-python will throw an error if we try to do this; the params that are optimized are protected and cannot be modified)

### CODE REVIEW

**Review by yuhong-sigopt — APPROVED** [review]

File: sigopt/xgboost/run.py [hunk 1]
```diff
@@ -12,13 +12,14 @@
 from ..log_capture import SystemOutputStreamMonitor
 from .. import create_run
 
-DEFAULT_EVALS_NAME = 'Test Set'
+DEFAULT_EVALS_NAME = 'TestSet'
```
Comment (harvey-intel) [inline sigopt/xgboost/run.py:15]: Should we also remove the space in `Training Set`

Comment (ericlee0803) [reply sigopt/xgboost/run.py:15]: good point, done

File: sigopt/xgboost/run.py [hunk 2]
```diff
@@ -133,9 +134,8 @@ def log_metadata(self):
   def log_params(self):
     for name in self.params.keys():
```
Comment (yuhong-sigopt) [inline sigopt/xgboost/run.py:135]: `for` is not needed.

File: sigopt/xgboost/run.py [hunk 3]
```diff
@@ -133,9 +134,8 @@ def log_metadata(self):
...
       if name not in PARAMS_LOGGED_AS_METADATA:
-        self.run.params.update({name: self.params[name]})
-
-    self.run.params.num_boost_round = self.num_boost_round
+        self.run.params.setdefaults(self.params)
+    self.run.params.setdefaults({'num_boost_round': self.num_boost_round})
```
Comment (harvey-intel) [inline sigopt/xgboost/run.py:138]: I thought `run.params.setdefaults` doesn't propagate to the web.

File: sigopt/xgboost/run.py (hunk 2 above)
Comment (harvey-intel) [reply sigopt/xgboost/run.py:135]: I think you want `self.run.params.setdeefaults({name: self.params[name]})`

**Review by harvey-intel — COMMENTED** [review]
just one minor error

**Comment by ericlee0803** [comment]
Will revisit once parameter sources is done.

**Comment by yuhong-sigopt** [comment]
Block it by waiting for parameter source is available. The user default parameters should be showed in web UI.

**Comment by ericlee0803** [comment]
This is an old PR that is now outdated.
